shell/probe/nocturnal.sh

```diff
@@ -1,30 +1,32 @@
 #!/bin/bash
 
 PRELUDE_API=${PRELUDE_API:="https://api.preludesecurity.com"}
+PRELUDE_DIR=${PRELUDE_DIR:="./cache"}
+PRELUDE_SLEEP=${PRELUDE_SLEEP:=14440}
 
-sys=$(uname -s)-$(uname -m)
-dos=$(echo $sys | tr '[:upper:]' '[:lower:]')
+dos=$(echo $(uname -s)-$(uname -m))
 
 while :
-do
-    temp=$(mktemp)
-    location=$(curl -sL -w %{url_effective} -o $temp -H "token:${PRELUDE_TOKEN}" -H "dos:${dos}" -H "dat:${dat}" $PRELUDE_API)
-    test=$(echo $location | grep -o '[0-9a-f]\{8\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{12\}' | head -n 1)
+do  
+    exe=$PRELUDE_DIR/$(uuidgen)
+    location=$(curl -sL -w %{url_effective} --create-dirs -o $exe -H "token:${PRELUDE_TOKEN}" -H "dos:${dos}" -H "dat:${dat}" $PRELUDE_API)
+    test=$(echo $location | grep -o '[0-9a-f]\{8\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{12\}' | head -n 1)    
 
-    if [ -z "$test" ];then
-        sleep 14400
-    else
+    if [ $test ];then
         ca=$(echo $location | sed -e 's|^[^/]*//||' -e 's|/.*$||')
-        if [ -z "$PRELUDE_CA" ] || [ "$PRELUDE_CA" == "$ca" ];then
-            chmod +x $temp
 
-            if test -f "$temp";then
-                $temp
-                dat="${test}:$?"
-                $temp -cleanup
-            else
-                dat="${test}:9"
+        if [ -z "$PRELUDE_CA" ] || [ "$PRELUDE_CA" == "$ca" ];then
+            chmod +x $exe
+            
+            if test -f $exe;then
+                echo "[P] Running $test masquerading as $exe"
+                $exe
             fi
+            dat="${test}:$?"
         fi
+    else
+        echo "[P] Cleaning up temporary cache"
+        find $PRELUDE_DIR -type f -name "*" -mmin -1 -delete
```

Comment:
don't really need the `-name *` anymore

this doesn't seem particularly scalable (our tests timeout at 10s so if someones setup sucks, they might not finish all tests in 1min. Especially as we keep adding more tests)

also a little scary to "delete all", esp. for a directory as vaguely named as "cache"

Reply:
yeah; I've been working on this area a bit. going between a few thoughts atm.